net/isc-dhcp/files/dhcpd.init

```diff
@@ -127,14 +126,90 @@ rev_str() {
 	echo "$result"
 }
 
-create_empty_zone() {
-	local zone="$1"
+create_zone_update_file() {
+	local zone=$1
+	local zonefile="$dyndir/db.$zone"
 
-	if [ ! -f $dyndir/db."$zone" ]; then
-		cp -p /etc/bind/db.empty $dyndir/db."$zone"
-		chmod g+w $dyndir/db."$zone"
-		chgrp bind $dyndir/db."$zone"
-	fi
+	cp -p /etc/bind/db.empty "$zonefile"
+
+	cat <<EOF >> $conf_local_file
+zone "$zone" {
+	type master;
+	file "$zonefile";
+	update-policy {
+		grant $session_key_name zonesub any;
+	};
+};
+EOF
+}
+
+print_dhcp_ddns_config() {
+	local zone=$1
+	local cfg=$2
+	local server
+
+	config_get server "$cfg" server
+
+	cat <<EOF
+zone $zone. {
+	primary $server;
+	key $session_key_name;
+}
+EOF
+}
+
+generate_session_file() {
+	local cfg=$1
+
+	config_get session_key_name $cfg key_name local-dns
+	config_get key_algo $cfg key_algo hmac-sha256
+	config_get key_secret $cfg key_secret
+
+	cat <<EOF > "$session_key_file"
+key "$session_key_name" {
+	algorithm $key_algo;
+	secret "$key_secret";
+};
+EOF
+}
+
+generate_ddns_config() {
+	local domain=$1
+	local cfg=dynamicdns
+	local server zones
+
+	config_get server $cfg server
+	config_get zones $cfg zones
+
+	generate_session_file "$cfg"
+
+	cat <<EOF
+ddns-domainname "$domain.";
+ddns-update-style standard;
+ddns-updates on;
+ignore client-updates;
+
+update-static-leases on;
+use-host-decl-names on;
+update-conflict-detection off;
+update-optimization off;
+
+include "$session_key_file";
+
+EOF
+
+	config_list_foreach $cfg zones create_zone_update_file "$cfg"
+	config_list_foreach $cfg zones print_dhcp_ddns_config "$cfg"
+
+	chmod g+w "$dyndir" -R
+	chgrp bind "$dyndir" -R
+
+	/etc/init.d/named reload
```

Comment:
Rebased code with the master branch. 

But there seems to be some issue in that code.

```bash
		cp -p $conf_local_file ${conf_local_file}_
		cmp -s $conf_local_file ${conf_local_file}_ || need_reload=1
		rm -f ${conf_local_file}_
```

Here if first copies the file and then compares, in this case, both files would be always the same. Am I missing anything there?

Reply:
No, that's a bug.  Fixed in PR #19866

Reply:
Fix here...?

Reply:
It would be picked up during a rebase.

Reply:
> I've added some code (as PR #19532) where we save the existing `/tmp/bind/named.conf.local` before rewriting it, then only do the `reload` if the new version has changed.
> 
> Forcing a reload causes the in-memory journal to get stomped on by the zone reload from the file.

I had the thought that this might not actually be a bad thing. It's nice to know which change it was exactly that caused something to go bang, rather than e.g. several config iterations go in, same cache is kept, then the 8th change makes it go bang. Some users may be inclined to suspect that 1-7 may have compounded a hidden error. 

At least this is a thought process I sometimes find myself in when I'm 'bisecting' config problems. The quickest way out of this, however, is to roll back to the previous config, if available.

Perhaps there is merit to retain the last known working config.